wgpu/src/backend/web.rs

```diff
@@ -1914,6 +1914,11 @@ impl crate::context::Context for Context {
         // Device is dropped automatically
     }
 
+    fn device_lose(&self, _device: &Self::DeviceId, _device_data: &Self::DeviceData) {
+        // TODO: figure out the GPUDevice implementation of this, including resolving
+        // the device.lost promise.
```

Comment:
since WebGPU doesn't expose this, there won't be ever any implementation for this (see also comment on wgpu/src/lib.rs and tests). Exposing the promise we need to do seperatly, likely similar to how we expose other async operations like buffer mapping.

Reply:
I modified the TODO comment to better state the next steps.